### ISSUE TITLE
halyard version 1.22.0 ! ERROR Could not translate your halconfig: Unrecognized field

### ISSUE THREAD

**Issue by roma-d** [spinnaker/spinnaker#4661]
### Issue Summary:
When I try to add a new Kubernetes account, halyard version 1.22.0 throws the following error:

```
spinnaker@6b257c00b343:/workdir$ hal config provider kubernetes account add test-k8s \
>            --provider-version v2 \
>            --context $CONTEXT
- Get current deployment
  Failure
Problems in Global:
! ERROR Could not translate your halconfig: Unrecognized field
  "oauthScopes" (class
  com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesAccount),
  not marked as ignorable (30 known properties: "permissions", "cacheThreads",
  "name", "oAuthServiceAccount", "environment", "requiredGroupMembership",
  "liveManifestCalls", "dockerRegistries", "kinds", "omitNamespaces",
  "checkPermissionsOnStartup", "providerVersion", "onlySpinnakerManaged",
  "configureImagePullSecrets", "context", "customResources", "oAuthScopes",
  "serviceAccount", "cluster", "skin", "user", "kubectlPath", "omitKinds",
  "namingStrategy", "kubectlRequestTimeoutSeconds", "kubeconfigContents",
  "cachingPolicies", "namespaces", "kubeconfigFile", "debug"])
at [Source: UNKNOWN; line: -1, column: -1] (through reference chain:
  com.netflix.spinnaker.halyard.config.model.v1.node.Halconfig["deploymentConfigurations"]->java.util.ArrayList[0]->com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration["providers"]->com.netflix.spinnaker.halyard.config.model.v1.node.Providers["kubernetes"]->com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesProvider["accounts"]->java.util.ArrayList[0]->com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesAccount["oauthScopes"])

- Failed to get deployment name.
```
It works fine with the previous version 1.21.2
### Cloud Provider(s):
Kubernetes

### Environment:
halyard docker container

### Additional Details:
It looks like there's a rename from `oauthScopes: []` to `oAuthScopes: []` that creates this issue. I removed the `oauthScopes: []` part from `~/.hal/config` and it is working fine afterwards

**Comment by maggieneterval** [comment]
Hey @roma-d, thanks so much for opening this issue! There is a 
[Slack discussion](https://app.slack.com/client/T091CRSGH/C3YBRLG8H/thread/C3YBRLG8H-1563564697.170400) happening about this very problem: looks like the running theory is that somehow both `oauthScopes` and `oAuthScopes` had been added to configs at some point, and only after the spring boot 2 upgrade did Halyard throw an error. Your workaround of removing `oauthScopes` is the best path forward for now!

**Comment by shaffi** [comment]
@maggieneterval  Thanks, Had the same issue, removing `oauthScopes ` from `~/.hal/config` worked.

**Comment by plumpy** [comment]
This is fixed in 1.22.1 by spinnaker/halyard#1376.
